### PR TITLE
PR to check the direction.

### PR DESCRIPTION
**Pull request by mikhail-akimov**
Am i in a right direction? Or i should to specify the defaults to forbid?

And im not sure about the description of `ComplexDefaultValuesViolation`... I`ll fix it after i complete the `_check_argument_default_value`.

### CODE REVIEW

File: wemake_python_styleguide/visitors/ast/functions.py [hunk 1]
```diff
@@ -149,6 +150,22 @@ def _check_unused_variables(self, node: AnyFunctionDef) -> None:
                 )
         self._check_used_variables(local_variables)
 
+    def _check_argument_default_values(self, node: AnyFunctionDef) -> None:
+
+        for arg in node.args.defaults:
+            if not isinstance(arg, (
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/functions.py]: please, move this tuple to the class level, the same as here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/ast/statements.py#L66-L71

File: wemake_python_styleguide/violations/best_practices.py [hunk 2]
```diff
@@ -1538,7 +1540,8 @@ class TryExceptMultipleReturnPathViolation(ASTViolation):
 @final
 class ComplexDefaultValuesViolation(ASTViolation):
     """
-    Forbids to use anything that is not a ``ast.Name`` or ``ast.Attribute`` as
+    Forbids to use anything that is not a `ast.Name`, `ast.Attribute`,
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py]: We use double ` chars to indicate that it is a code.
That's how sphinx renders it.

**Comment by mikhail-akimov** [comment]
I`ll fix it :) 
I forgot to check best_practices.py :(

**Comment by mikhail-akimov** [comment]
I`m stuck. Could you help me to understand what exactly happening in this test-case? Or may be you could get me some hint how to debug it? I don`t understand what happening in ``process.communicate()``, but can assume that this func. returns list of violations indexes.

And when i starts "poetry run pytest -v tests\test_checker\test_noqa.py" i get this error:

 def _load_all_violation_classes():
        modules = [
>           violations.naming,
            violations.complexity,
            violations.consistency,
            violations.best_practices,
        ]
E       AttributeError: module 'wemake_python_styleguide.violations' has no attribute 'naming'

tests\conftest.py:37: AttributeError

**Comment by sobolevn** [comment]
@mikhail-akimov you need to update `tests/fixtures/noqa.py` file with a new violation example.

**Comment by sobolevn** [comment]
Check https://github.com/wemake-services/wemake-python-styleguide/blob/master/CONTRIBUTING.md#before-submitting

**Comment by mikhail-akimov** [comment]
I still can not understand... I have other errors locally.

File: tests/fixtures/noqa.py [hunk 3]
```diff
@@ -345,3 +345,11 @@ def multiple_return_path():
         return 2
     else:
         return 3
+
+
+def bad_default_values(
+    bad,
+    defaults=(),
```
Comment (sobolevn) [inline tests/fixtures/noqa.py]: These values are not bad. We allow strings, tuples, and other primitives. 
You need to provide wrong values here.

Comment (mikhail-akimov) [reply tests/fixtures/noqa.py]: Oh, my bad! Thank you!

File: tests/fixtures/noqa.py [hunk 4]
```diff
@@ -348,8 +348,9 @@ def multiple_return_path():
...
-    bad,
-    defaults=(),
-    shall='notpass',
+    self,
+    filename='(none)',
+    builtins=None,
+    withDoctest='PYFLAKES_DOCTEST' in os.environ,
 ):
-    return True  # noqa: Z459
+    return True # noqa: Z459
```
Comment (sobolevn) [inline tests/fixtures/noqa.py]: It should be: `def bad_default_values(  # noqa: Z459`

Comment (mikhail-akimov) [reply tests/fixtures/noqa.py]: Why? How it works?

**Comment by sobolevn** [comment]
Thanks @mikhail-akimov for the awesome work you have done!
Feel free to choose more issues in the future. This project needs your help!

**Comment by mikhail-akimov** [comment]
Thank you! 

I need some time for a retrospective, and then come back for the next issue.

**Pull request merged**